Interesting - `umap-learn` (both py27 and py37) on macos 10.13 gives me

```
running install tests
....................................................F.......................
======================================================================
FAIL: umap.tests.test_umap.test_umap_transform_on_iris
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/nix/store/7zlrb2lkq5rv24x4a60vjpli2b9hl712-python3.7-nose-1.3.7/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/private/tmp/nix-build-python3.7-umap-learn-0.3.9.drv-0/source/umap/tests/test_umap.py", line 767, in test_umap_transform_on_iris
    trust),
AssertionError: 0.889159891598916 not greater than or equal to 0.89 : Insufficiently trustworthy transform foriris dataset: 0.889159891598916
```

Which... I'm not sure what to do about 🤷‍♂

--- a/pkgs/development/python-modules/kmapper/default.nix
+++ b/pkgs/development/python-modules/kmapper/default.nix
@@ -1,0 +1,51 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, scikitlearn
+, numpy
+, scipy
+, jinja2
+, pytest
+, networkx
+, matplotlib
+, python-igraph
+, plotly
+, ipywidgets
+}:
+
+buildPythonPackage rec {
+  pname = "kmapper";
+  version = "1.2.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0879294680c5d01a928847b818a3c4e07eded3f602f96e510858e68e74fa3783";
+  };
+
+  propagatedBuildInputs = [
+    scikitlearn
+    numpy
+    scipy
+    jinja2
+  ];
+
+  checkInputs = [
+    pytest
+    networkx
+    matplotlib
+    python-igraph
+    plotly
+    ipywidgets
+  ];
+
+  checkPhase = ''
+    pytest test --ignore test/test_drawing.py
+  '';
+
+  meta = with lib; {
+    description = "Python implementation of Mapper algorithm for Topological Data Analysis";
+    homepage = http://kepler-mapper.scikit-tda.org;
+    license = licenses.mit;
+    maintainers = [ maintainers.costrouc ];
+  };
+}

--- a/pkgs/development/python-modules/ripser/default.nix
+++ b/pkgs/development/python-modules/ripser/default.nix
@@ -1,0 +1,48 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, cython
+, numpy
+, scipy
+, scikitlearn
+, persim
+, pytest
+}:
+
+buildPythonPackage rec {
+  pname = "ripser";
+  version = "0.3.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "ff9f50fba911f0e9212077b78014f83e30c97526194dd6bd1df3d81896e6cb58";
+  };
+
+  checkInputs = [
+    pytest
+  ];
+
+  propagatedBuildInputs = [
+    cython
+    numpy
+    scipy
+    scikitlearn
+    persim
+  ];
+
+  checkPhase = ''
+    # specifically needed for darwin
+    export HOME=$(mktemp -d)
+    mkdir -p $HOME/.matplotlib
+    echo "backend: ps" > $HOME/.matplotlib/matplotlibrc
+
+    pytest
+  '';
+
+  meta = with lib; {
+    description = "A Lean Persistent Homology Library for Python";
+    homepage = https://ripser.scikit-tda.org;
+    license = licenses.mit;
+    maintainers = [ maintainers.costrouc ];
+  };
+}

--- a/pkgs/development/python-modules/scikit-tda/default.nix
+++ b/pkgs/development/python-modules/scikit-tda/default.nix
@@ -1,0 +1,64 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, numpy
+, scipy
+, scikitlearn
+, matplotlib
+, numba
+, umap-learn
+, cython
+, ripser
+, persim
+, pillow
+, kmapper
+, tadasets
+, pytest
+, isPy27
+}:
+
+buildPythonPackage rec {
+  pname = "scikit-tda";
+  version = "0.0.4";
+  disabled = isPy27;
+
+  src = fetchFromGitHub {
+    owner = "scikit-tda";
+    repo = "scikit-tda";
+    rev = version;
+    sha256 = "0a90k6i9fkmc9gf250b4fidx2fzd2qrn025l74mjk51fvf23q13a";
+  };
+
+  propagatedBuildInputs = [
+    numpy
+    scipy
+    scikitlearn
+    matplotlib
+    numba
+    umap-learn
+    cython
+    ripser
+    persim
+    pillow
+    kmapper
+    tadasets
+  ];
+
+  checkInputs = [
+    pytest
+  ];
+
+  checkPhase = ''
+    pytest test
+  '';
+
+  # tests will be included in next release
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Topological Data Analysis for humans";
+    homepage = https://github.com/scikit-tda/scikit-tda;
+    license = licenses.mit;
+    maintainers = [ maintainers.costrouc ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -4435,6 +4435,8 @@ in {
 
   scikit-optimize = callPackage ../development/python-modules/scikit-optimize { };
 
+  scikit-tda = callPackage ../development/python-modules/scikit-tda { };
+
   scp = callPackage ../development/python-modules/scp {};
 
   seaborn = callPackage ../development/python-modules/seaborn { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -4314,6 +4314,8 @@ in {
     python2 = python;
   }));
 
+  ripser = callPackage ../development/python-modules/ripser { };
+
   rjsmin = callPackage ../development/python-modules/rjsmin { };
 
   pysolr = callPackage ../development/python-modules/pysolr { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -3827,6 +3827,8 @@ in {
 
   pika-pool = callPackage ../development/python-modules/pika-pool { };
 
+  kmapper = callPackage ../development/python-modules/kmapper { };
+
   kmsxx = (callPackage ../development/libraries/kmsxx {
     inherit (pkgs.kmsxx) stdenv;
     inherit (pkgs) pkgconfig;

